api-error-response.go

```diff
@@ -83,6 +83,32 @@ const (
 	reportIssue = "Please report this issue at https://github.com/minio/minio-go/issues."
 )
 
+type APIErrorCode string
+
+var (
+	ErrCodeNoSuchBucket          = APIErrorCode("NoSuchBucket")
```

Comment:
Currently, developers need to read  minio-go API + S3 spec to understand errors received from minio-go library. I just wanted to make them look only at minio-go documentation with some details of the errors that they could receive. The code will still work if we receive an error code that we haven't seen before. In that case, I think the only change that we can do is adding a new exported error variable as I did above and this will be a backward compatible change. 



> Minor deviations require significant amount of code change on our end.

Can you elaborate on this?